pkg/hooks/hooks_test.go

```diff
@@ -77,3 +79,62 @@ func testManifest() pods.Manifest {
 	builder.SetID(podId)
 	return builder.GetManifest()
 }
+
+func TestHookRunWithTimeout(t *testing.T) {
+	type sleepTest struct {
+		sleep      int
+		timeout    int
+		shouldPass bool
+	}
+
+	var sleepTests = []sleepTest{
+		{1, 2, true},
+		{2, 1, false},
```

Comment:
this will be kind of a long running test, i wonder if there's a way to cut it down. Maybe if you abstract out the use of `time.After` as the channel selected on you can just write immediately do it